docs/dev/conventions/naming_conventions.md

```diff
@@ -1,11 +1,29 @@
-Casing Types:
-- [Kebab-case](https://en.wikipedia.org/wiki/Kebab_case)
-
 ## Java
 
-- In general, use the naming conventions from the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html#s5-naming).
-- Use a `new` prefix for factory methods (e.g. `newFoo()` instead of `createFoo()`).
+- Follow the naming in: [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html#s5-naming)
+- Use a `new` prefix for factory methods (e.g. `newFoo()` instead of `createFoo()`)
 
 ## URLs
 
-- kebab-case
+- [kebab-case](https://en.wikipedia.org/wiki/Kebab_case)
+
+
+## Database
+
+- [snake_case](https://en.wikipedia.org/wiki/Snake_case) 
+- Use singular names, eg: `user` instead of `users`
```

Comment:
I was a bit on the fence with singular vs plural naming. Some quick googling and it seems like there is some slight bias towards singular, but not to hard to find it is a split opinion.

I thought this was a pretty nice write-up: https://www.databasestar.com/database-table-naming-conventions/

For a TripleA context, I like in the example of a 'user' table the name of the table does not change when used in join tables. eg: `user_ban` instead of `users_ban` or `banned_users`

Reply:
I'm all for the singular variant.
Also regarding bans and mutes, we probably are fine with a single punishment table and some sort of enum to determine the type of punishment

Reply:
Also for me it makes more sense to write something like `WHERE user.id = 1234` in queries

Reply:
Ah, it finally dawned on me that it is the caps in `WHERE` that you were pointing out @RoiEXLab , I was focused on the conditional part :grin: 

The all lower_snake is a bit intentional, with hopefully some benefits:
- the global rule, all lower_snake on all names/things DB is a helpful consistency
- perhaps subjective, the important of queries are not the keywords, but the table and column names and conditions. Putting the keywords in caps is not necessary helpful (unless one is learning the keywords, in which case it perhaps is).

It's kinda subjective either way, I land more towards all lower snake.

Reply:
Actually no 😅 This isn't what I meant.
I was just trying to make another point for singular vs plural naming.

About the caps thingy: I still like writing the important keywords in caps, but I accepted that this project wants a different style guide.